lib/EnsEMBL/REST/Controller/Sequence.pm

```diff
@@ -265,6 +308,119 @@ sub _process_feature {
   return \@sequences;
 }
 
+# For grabbing the sub-sequence if requested
+# The start and end coordinates come from the user request
+sub _process_subseq {
+  my ($self, $c, $seq) = @_;
+
+  my $seq_len = length $seq;
+
+  # Get our start/end points and do some sanity checking
+  my ($start, $end) = $self->_check_limits($c, $seq_len);
+
+  # Deal with zero indexing, we do the minus one here rather than the
+  # sanity checking routine so the sanity checking routine can be
+  # reused for the Slice() execution path where we need a default
+  # start of 1.
+  return substr $seq, $start-1, $end-$start+1;
+}
+
+# Check the trimming start/end we're given by the user to ensure
+# they make sense, and fill in sensible default if one is left out.
+sub _check_limits {
+  my ($self, $c, $seq_len) = @_;
+
+  # Oh the fun of zero indexing strings in a one-indexed sub-sequence world
+  my $start = $c->request->param('start') ? $c->request->param('start') : 1;
+  my $end = defined($c->request->param('end')) ? $c->request->param('end') : $seq_len;
+
+  # Sanity checking
+  if($start < 0 || $start > $seq_len) {
+      Catalyst::Exception->throw(qq{Your start coordinate is not within the sequence requested})
+  }
+  if($end < 1 || $end > $seq_len + 1) {
+      Catalyst::Exception->throw(qq{Your end coordinate is not within the sequence requested})
+  }
+  if($start > $end) {
+      Catalyst::Exception->throw(qq{Your start coordinate cannot be larger than your end})
```

Comment:
Yes, it should take into account circular genomes as well, in principle. My work on them in the past was mainly about making sure the features crossing the origin had the correct coordinates wrt their associated slice. I didn't thoroughly test the sequence aspect of circular genomes. I'd leave Matthew constraint at the moment, and see if EG is interested in the exception.